src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c

```diff
@@ -551,7 +551,7 @@ int32_t GlobalizationNative_LastIndexOf(
         result = GlobalizationNative_CompareString(pSortHandle, lpTarget, cwTargetLength, lpSource, cwSourceLength, options);
         if (result == UCOL_EQUAL && pMatchedLength != NULL)
         {
-            *pMatchedLength = cwTargetLength;
+            *pMatchedLength = cwSourceLength;
```

Comment:
This was incorrectly returning the target length instead of the source length and was causing some newly introduced unit tests to fail.